### PR TITLE
Add a script to transform galaxy.yml

### PR DESCRIPTION
**Pull request by pcahyna**
Replaces collection namespace, name and version, using ruamel.yaml

### CODE REVIEW

**Comment by i386x** [comment]
@pcahyna I think it needs `six` in `deps` [here](https://github.com/linux-system-roles/auto-maintenance/blob/master/tox.ini#L34).

**Comment by pcahyna** [comment]
> @pcahyna I think it needs `six` in `deps` [here](https://github.com/linux-system-roles/auto-maintenance/blob/master/tox.ini#L34).

@i386x thanks, that should be unrelated to my code though.

**Comment by nhosoi** [comment]
> > @pcahyna I think it needs `six` in `deps` [here](https://github.com/linux-system-roles/auto-maintenance/blob/master/tox.ini#L34).
> 
> @i386x thanks, that should be unrelated to my code though.

Sorry, @pcahyna. I've opened a pr just to add "six". Could you review it? And if possible, rerun the failed tox tests in this pr?
https://github.com/linux-system-roles/auto-maintenance/pull/47
In addition to this fix by @richm, the default tox.ini for auto-maintenance needed the same fix...
https://github.com/linux-system-roles/tox-lsr/pull/19

**Review by richm — DISMISSED** [review]
lgtm

**Comment by nhosoi** [comment]
Thank you, @richm. https://github.com/linux-system-roles/auto-maintenance/pull/47 has been merged.

**Review by nhosoi — APPROVED** [review]
lgtm

**Comment by nhosoi** [comment]
@pcahyna,
I ran the script like this:
python galaxy_transform.py "%{collection_namespace}" "%{collection_name}" "%{collection_version}" > galaxy.yml.tmp
And here's the result.
```
--- galaxy.yml	2021-02-09 13:10:51.000000000 -0800
+++ galaxy.yml.tmp	2021-02-09 15:50:16.150007766 -0800
@@ -1,18 +1,18 @@
-namespace: "fedora"
-name: "linux_system_roles"
-version: "0.0.2"
-description: "Ansible roles for linux system components management"
+namespace: redhat
+name: rhel_system_roles
+version: 1.0.0
+description: Ansible roles for linux system components management
 
 authors:
-  - "System Roles Owners <systemroles@lists.fedorahosted.org>"
+- System Roles Owners <systemroles@lists.fedorahosted.org>
 
-repository: "https://github.com/linux-system-roles/linux-system-roles.github.io/"
-documentation: "https://github.com/linux-system-roles/linux-system-roles.github.io/blob/master/README.md"
-homepage: "https://linux-system-roles.github.io"
-issues: "https://github.com/linux-system-roles/linux-system-roles.github.io/issues"
+repository: https://github.com/linux-system-roles/linux-system-roles.github.io/
+documentation: https://github.com/linux-system-roles/linux-system-roles.github.io/blob/master/README.md
+homepage: https://linux-system-roles.github.io
+issues: https://github.com/linux-system-roles/linux-system-roles.github.io/issues
 
-readme: "README.md"
+readme: README.md
 license:
     - GPL-2.0-or-later
     - GPL-3.0-only
     - GPL-3.0-or-later
@@ -22,6 +22,6 @@
 
 dependencies:
 
 tags:
-  - "system"
-  - "collection"
+- system
+- collection
```
Can we keep the indentation? And preferably, the double quotes, as well?

File: galaxy_transform.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/python3
+
+import sys
+from ruamel.yaml import YAML
+
+filepath = "galaxy.yml"
+
+buf = open(filepath).read()
+
+yaml = YAML()
```
Comment (richm) [inline galaxy_transform.py:10]: try `yaml = YAML(typ="rt")`
That might help preserve quoting, indentation, etc.

Comment (richm) [reply galaxy_transform.py:10]: well, you need that, and this:
```python
yaml.default_flow_style = False
yaml.preserve_quotes = True
yaml.indent(mapping=2, sequence=4, offset=2)
```

Comment (richm) [reply galaxy_transform.py:10]: That gives the following:
```yaml
namespace: "mynamespace"
name: "myname"
version: "5.4.3"
description: "Ansible roles for linux system components management"

authors:
  - "System Roles Owners <systemroles@lists.fedorahosted.org>"

repository: "https://github.com/linux-system-roles/linux-system-roles.github.io/"
documentation: "https://github.com/linux-system-roles/linux-system-roles.github.io/blob/master/README.md"
homepage: "https://linux-system-roles.github.io"
issues: "https://github.com/linux-system-roles/linux-system-roles.github.io/issues"

readme: "README.md"
license:
  - GPL-2.0-or-later
  - GPL-3.0-only
  - GPL-3.0-or-later
  - LGPL-3.0-only
  - MIT
  - BSD-3-Clause

dependencies:

tags:
  - "system"
  - "collection"
```

Comment (nhosoi) [reply galaxy_transform.py:10]: No luck so far...

My ruamel.yaml version is 0.16.12.

Comment (nhosoi) [reply galaxy_transform.py:10]: Please note that it is not a critical issue. There were no errors nor even a warning in uploading to the test automation-hub with the following galaxy.yml. So, we don't have to worry about it, I guess. I verified the generated MANIFEST.json looks valid.

I'm going to approve this pr again.
```
namespace: redhat
name: rhel_system_roles
version: 1.0.10
description: Ansible roles for linux system components management

authors:
- System Roles Owners <systemroles@lists.fedorahosted.org>

repository: https://github.com/linux-system-roles/linux-system-roles.github.io/
documentation: https://github.com/linux-system-roles/linux-system-roles.github.io/blob/master/README.md
homepage: https://linux-system-roles.github.io
issues: https://github.com/linux-system-roles/linux-system-roles.github.io/issues

readme: README.md
license:
- GPL-2.0-or-later
- GPL-3.0-only
- GPL-3.0-or-later
- LGPL-3.0-only
- MIT
- BSD-3-Clause

dependencies:

tags:
- system
- collection
 ```

**Review by nhosoi — APPROVED** [review]

File: galaxy_transform.py (hunk 1 above)
Comment (nhosoi) [reply galaxy_transform.py:10]: Thank you, @richm! I missed your comments while testing with automation-hub. :p

I verified your proposed code generates the valid galaxy.yml. I'd like to include your changes in this pr. > @pcahyna

Comment (pcahyna) [reply galaxy_transform.py]: I added new commits with the proposed changes.

**Review by nhosoi — APPROVED** [review]
lgtm

**Comment by nhosoi** [comment]
Sorry, @pcahyna.
I tested the latest auto-maintenance with this pr and noticed if there is an empty value ("dependencies:" in galaxy.yml), ansible-generator fails as follows in the packaging. Could you please review this [pr](https://github.com/linux-system-roles/auto-maintenance/pull/48) to fix it? Thanks!!
```
Traceback (most recent call last):
  File "/usr/lib/rpm/ansible-generator", line 46, in <module>
    main()
  File "/usr/lib/rpm/ansible-generator", line 31, in main
    for dep, req in info.get("dependencies", {}).items():
AttributeError: 'NoneType' object has no attribute 'items
```'

**Comment by pcahyna** [comment]
> Sorry, @pcahyna.
> I tested the latest auto-maintenance with this pr and noticed if there is an empty value ("dependencies:" in galaxy.yml), ansible-generator fails as follows in the packaging. Could you please review this [pr](https://github.com/linux-system-roles/auto-maintenance/pull/48) to fix it? Thanks!!
> 
> ```
> Traceback (most recent call last):
>   File "/usr/lib/rpm/ansible-generator", line 46, in <module>
>     main()
>   File "/usr/lib/rpm/ansible-generator", line 31, in main
>     for dep, req in info.get("dependencies", {}).items():
> AttributeError: 'NoneType' object has no attribute 'items
> ```'
> ```

How exactly did you test? IIRC, ansible-generator is used only on Fedora. But on Fedora, we should not be transforming galaxy.yml in the first place, we should use the original from this repo, unmodified (there is no need to replace namespace & collection name on Fedora...).

**Pull request merged**